### PR TITLE
Add NeoForge support

### LINKED ISSUE #1356: Add support for NeoForge modloader
### Role

I play modded minecraft and run modded servers.

### Suggestion

Add support for the Neo Forged modloader (a hard fork of Forge including all the team excluding LexManos)

### Benefit

Neo Forged will be the only replacement for Forge unless LexManos gets a whole new team or someone else decides to fork it.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

The repo for neo forged is here: https://github.com/neoforged/NeoForge they will contain 100% comparability with Forge until the next minecraft release then then they're going to break hard, to fix all of the things that should have been fixed years ago.

### PR DESCRIPTION
**Pull request by Scrumplex**
Needs PrismLauncher/meta#21

closes #1356 

Also waiting for feedback on modrinth/docs#118

### CODE REVIEW

**Comment by TheKodeToad** [comment]
I don't think we should merge this until NeoForge is "ready" (whatever that could be), therefore it's probably best to put the effort into fixing conflicts instead of merging to staging?

**Comment by Trial97** [comment]
Should add neoforge to flame instance creation/exporting  in a separate PR?

**Comment by Scrumplex** [comment]
> Should add neoforge to flame instance creation/exporting in a separate PR?

I wasn't sure how it's gonna work so I left it out for now.

Let's get this merged and in develop first! :D

**Comment by Trial97** [comment]
That sounds great!
Hope it's as easy as adding quilt support.
Also here is a possible implementation for instance creation: https://github.com/PrismLauncher/PrismLauncher/compare/develop...Nanite-Prism:PrismLauncher:develop#diff-15705752afc579f76953b4c0ffb23fa5b24dfb45a1b97b330d36c46f195b9d94R364-R367

**Comment by Ryex** [comment]
heads up that curseforge already supports neoforge. it's ClassID is 6

**Comment by Scrumplex** [comment]
> heads up that curseforge already supports neoforge. it's ClassID is 6

I added this already :D

File: launcher/modplatform/ResourceAPI.h [hunk 1]
```diff
@@ -54,7 +54,7 @@ class ResourceAPI {
    public:
     virtual ~ResourceAPI() = default;
 
-    enum ModLoaderType { Forge = 1 << 0, Cauldron = 1 << 1, LiteLoader = 1 << 2, Fabric = 1 << 3, Quilt = 1 << 4 };
+    enum ModLoaderType { Forge = 1 << 0, Cauldron = 1 << 1, LiteLoader = 1 << 2, Fabric = 1 << 3, Quilt = 1 << 4, NeoForge = 1 << 5 };
```
Comment (TheKodeToad) [inline launcher/modplatform/ResourceAPI.h]: Why can't this be reshuffled?

Comment (Ryex) [reply launcher/modplatform/ResourceAPI.h]: if it's not serialized numerically anywhere it can be.

Comment (TheKodeToad) [reply launcher/modplatform/ResourceAPI.h]: ~~so i'm guessing it can be~~
I think you missed a comma and I misunderstood

Comment (Trial97) [reply launcher/modplatform/ResourceAPI.h]: It doesn't look like we serialize it numerically(from a quick search in the code).

**Comment by TheKodeToad** [comment]
Oops, I forgot about the meta pr

**Review by Trial97 — APPROVED** [review]

**Comment by Trial97** [comment]
Leaving the patch for adding neoforge to curseforge here: https://github.com/Scrumplex/PrismLauncher/compare/neoforge...Trial97:PrismLauncher:curse_neoforge?expand=1

In case you do not integrate it with this PR I will open one after this one is merged.
Disclaimer: I'm yet to test the proposed changes as I do not have the meta information

**Comment by TheKodeToad** [comment]
CurseForge doesn't support nf modpacks yet afaik

**Comment by Trial97** [comment]
> CurseForge doesn't support nf modpacks yet afaik

Yet they have a filter for it in the Web view:

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**